Libraries/Text/TextInput/RCTBaseTextInputShadowView.m

```diff
@@ -101,6 +102,17 @@ - (void)invalidateContentSize
   });
 }
 
+- (NSString *)text
+{
+  return _text;
+}
+
+- (void)setText:(NSString *)text
+{
+  _text = text;
+  _previousAttributedText = _localAttributedText;
```

Comment:
I think we simply have to call `[self dirtyLayout];` here.

Reply:
@shergin I've tried this out.  There is a still a bug where when iOS autocompletes it still doesn't clear.  Something happens with the autocomplete where it blocks the clear.

Reply:
That doesn't work.

Reply:
just to avoid any ambiguity:
`_previousAttributedText = _localAttributedText;` works

Reply:
@shergin @kevinresol any updates?

Reply:
@shergin you requested changes on this 10 weeks ago. Any chance you could approve it as is? This issue causes Android text boxes to be unusable. Its a major blocker for many production apps.

Reply:
This PR is waiting for @magicien to follow up to @shergin's feedback, or alternatively for @shergin to sign off on the PR to be merged as-is.

It's not clear to me that this can be merged just yet, as it seems like @shergin asked for `[self dirtyLayout]` to be called after `_previousAttributedText = _localAttributedText;` and the feedback has not been addressed.

Reply:
What I understand from @shergin 's comment is to replace `_previousAttributedText = _localAttributedText;` with `[self dirtyLayout];` and I have already stated clearly that it does not work.

Reply:
@shergin @kevinresol @hramos Apologies for taking so long to reply.
I would like to make sure what @shergin means.
@shergin, do you mean this code?

```
- (void)setText:(NSString *)text
{
  _text = text;
  _previousAttributedText = _localAttributedText;
  [self dirtyLayout];
}
```

I'm wondering if it needs to call `dirtyLayout` because `dirtyLayout` is called after `setText` is called. Calling `dirtyLayout` could be safer, though.
If the above code is the expected one, I will update this PR.

Reply:
Would be nice to have a reply/merge @shergin to resolve this. IMO, the PR is working as it is, and the suggested change doesn't seem to work.

Reply:
@shergin : a friendly reminder to review this, thanks.

Reply:
@magicien If you can confirm that `_previousAttributedText = nil;` works and update the diff, I would land this.